### PR TITLE
feat: GitHub actions

### PR DESCRIPTION
**Pull request by vonabarak**
Added testing with upload coverage to codecov.io and linters check

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          pip install poetry==1.1.6
+          poetry install
+          poetry config virtualenvs.in-project true
```
Comment (lxmnk) [inline .github/workflows/actions.yml:22]:
```suggestion
          poetry config virtualenvs.in-project true
          poetry install
```

Comment (lxmnk) [inline .github/workflows/actions.yml:22]:
```suggestion
          poetry config virtualenvs.in-project true
          poetry install
```

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: install dependencies
+        run: |
+          pip install poetry==1.1.6
+          poetry install
```
Comment (lxmnk) [inline .github/workflows/actions.yml:50]: Same here.

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          poetry config virtualenvs.in-project true
```
Comment (lxmnk) [inline .github/workflows/actions.yml:51]:
```suggestion
          poetry config virtualenvs.in-project true
          poetry install
```

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+name: Actions
+on: push
+jobs:
+
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
```
Comment (lxmnk) [inline .github/workflows/actions.yml:7]: Pls check can you reuse `runs-on`, `python-version` and `poetry-version` values in different jobs.

File: .github/workflows/actions.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+name: Actions
```
Comment (lxmnk) [inline .github/workflows/actions.yml:1]:
```suggestion
name: Python application
```

File: .github/workflows/actions.yml (hunk 4 above)
Comment (d-tamirlan) [reply .github/workflows/actions.yml:7]: And `Checkout` with `Set up Python`. Mb we can declare them once and then just reference them ?

File: .github/workflows/actions.yml (hunk 5 above)
Comment (vonabarak) [reply .github/workflows/actions.yml]: done

File: .github/workflows/actions.yml (hunk 1 above)
Comment (vonabarak) [reply .github/workflows/actions.yml]: done

File: .github/workflows/actions.yml (hunk 3 above)
Comment (vonabarak) [reply .github/workflows/actions.yml]: done

File: .github/workflows/actions.yml (hunk 4 above)
Comment (vonabarak) [reply .github/workflows/actions.yml:7]: Will be done in a separate PR.

**Review by d-tamirlan — APPROVED** [review]

**Pull request merged**